servo/connectors/opsani_dev.py

```diff
@@ -389,21 +440,21 @@ async def check_kubernetes_service_port(self) -> None:
 
         return f"Service Port: {port.name} {port.port}:{port.target_port}/{port.protocol}"
 
-    @servo.checks.check('Service routes traffic to Deployment Pods')
-    async def check_service_routes_traffic_to_deployment(self) -> None:
+    @servo.checks.check('Service routes traffic to {self.controller_type_name} Pods')
+    async def check_service_routes_traffic_to_controller(self) -> None:
         service = await servo.connectors.kubernetes.Service.read(
             self.config.service, self.config.namespace
         )
-        deployment = await servo.connectors.kubernetes.Deployment.read(
-            self.config.deployment, self.config.namespace
-        )
+        controller = await self.controller_class.read(self.config_controller_name, self.config.namespace)
+        # Remove label dynamically applied by argo-rollouts to service and rollout pods
+        service.selector.pop('rollouts-pod-template-hash', None)
```

Comment:
I don't love that Argo Rollouts specific stuff snuck in here when you almost have it fully abstracted. Why is this necessary?